crates/csslsrs/src/features/hover.rs

```diff
@@ -111,104 +112,171 @@ fn get_css_hover_content(
+fn format_css_property_entry(
     name: &str,
+    status: &Option<Status>,
     description: &Option<MarkupDescriptionOrString>,
     syntax: Option<&str>,
-    specificity: Option<(u32, u32, u32)>,
     browsers: Option<&[String]>,
     references: Option<&[Reference]>,
-    restrictions: Option<&[String]>,
 ) -> String {
     let mut content = String::new();
-    content.push_str(&format!("**{}**\n\n", escape_markdown(name)));
-
-    // Add the description if available
-    if let Some(description) = description {
-        match description {
-            MarkupDescriptionOrString::MarkupDescription(markup_description) => {
-                content.push_str(&markup_description.value);
-            }
-            MarkupDescriptionOrString::String(description) => {
-                content.push_str(description);
-            }
-        }
-        content.push_str("\n\n");
-    }
+    write_status(&mut content, status);
+    write_description(&mut content, description);
+    write_browser_support(&mut content, browsers);
+    write_syntax(&mut content, syntax);
+    write_references(&mut content, references, name);
+    content
+}
 
-    // Add syntax if available
-    if let Some(syntax) = syntax {
-        content.push_str(&format!("**Syntax**: `{}`\n\n", syntax));
-    }
+/// Formats the CSS at-rule entry into a hover content string.
+fn format_css_at_rule_entry(
+    name: &str,
+    status: &Option<Status>,
+    description: &Option<MarkupDescriptionOrString>,
+    references: Option<&[Reference]>,
+) -> String {
+    let mut content = String::new();
+    write_status(&mut content, status);
+    write_description(&mut content, description);
+    write_references(&mut content, references, name);
+    content
+}
 
+/// Formats the CSS selector entry into a hover content string.
+fn format_css_selector_entry(name: &str, specificity: Option<(u32, u32, u32)>) -> String {
+    let mut content = String::new();
+    // TODO: this is a placeholder, we should render an HTML preview of the selector
+    writeln!(content, "**{}**\n", escape_markdown(name)).unwrap();
     // Add specificity if available
     if let Some((ids, classes, elements)) = specificity {
-        content.push_str(&format!(
-            "[Selector Specificity](https://developer.mozilla.org/docs/Web/CSS/Specificity): ({}, {}, {})\n\n",
+        writeln!(content,
+            "[Selector Specificity](https://developer.mozilla.org/docs/Web/CSS/Specificity): ({}, {}, {})\n",
             ids, classes, elements
-        ));
+        ).unwrap();
     }
+    content
+}
 
-    // Add restriction if available
-    if let Some(restriction) = restrictions {
-        content.push_str("**Restriction**:\n");
-        for restriction in restriction {
-            content.push_str(&format!("- {}\n", restriction.trim()));
+/// Escapes Markdown characters in the selector name.
+fn escape_markdown(text: &str) -> String {
+    text.replace('*', "\\*")
```

Comment:
maybe pipes as well? Not sure if tables are supported in hover

Reply:
Not supported.